client/wsclient_test.go

```diff
@@ -59,6 +73,23 @@ func TestDisconnectWSByServer(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestDisconnectWSByClient(t *testing.T) {
+	// Start a Server.
+	srv := internal.StartMockServer(t)
+
+	defer srv.Close()
+
+	var settings types.StartSettings
+	settings.OpAMPServerURL = "ws://" + srv.Endpoint
+	client := NewWebSocket(TestLogger{t})
+	go func() {
+		time.Sleep(1 * time.Second)
+		startClient(t, settings, client)
+	}()
+	time.Sleep(2 * time.Second)
```

Comment:
Thanks @phanidevavarapu 

Is there any way we can wait for the specific event instead of sleeping for 1,2 seconds? This is fragile and also slows down tests.